API call to /usergroups  - Create user group should not fail with a 500 error

When trying to create a user group that already exists, harbor should not respond with an 500 error message and instead use a 409 to expose that an entry already exists.

```sh
2021-04-12T22:50:42Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"INTERNAL_SERVER_ERROR","message":"error occurred in add user group, error: pq: duplicate key value violates unique constraint \"unique_group_name\""}]}
/harbor/src/lib/http/error.go:82, github.com/goharbor/harbor/src/lib/http.apiError
/harbor/src/lib/http/error.go:51, github.com/goharbor/harbor/src/lib/http.SendError
/harbor/src/common/api/base.go:73, github.com/goharbor/harbor/src/common/api.(*BaseAPI).RenderError
/harbor/src/common/api/base.go:234, github.com/goharbor/harbor/src/common/api.(*BaseAPI).SendInternalServerError
/harbor/src/core/api/usergroup.go:157, github.com/goharbor/harbor/src/core/api.(*UserGroupAPI).Post
/harbor/src/vendor/github.com/astaxie/beego/router.go:835, github.com/astaxie/beego.(*ControllerRegister).ServeHTTP
```

Tested on Harbor 2.2.0 and 2.2.1

--- a/src/controller/member/controller.go
+++ b/src/controller/member/controller.go
@@ -68,10 +68,10 @@ type UserGroup struct {
 }
 
 // ErrDuplicateProjectMember ...
-var ErrDuplicateProjectMember = errors.New("The project member specified already exist")
+var ErrDuplicateProjectMember = errors.ConflictError(nil).WithMessage("The project member specified already exist")
 
 // ErrInvalidRole ...
-var ErrInvalidRole = errors.New("Failed to update project member, role is not in 1,2,3")
+var ErrInvalidRole = errors.BadRequestError(nil).WithMessage("Failed to update project member, role is not in 1,2,3")
 
 type controller struct {
 	userManager user.Manager

--- a/src/pkg/usergroup/dao/dao.go
+++ b/src/pkg/usergroup/dao/dao.go
@@ -56,7 +56,7 @@ func New() DAO {
 }
 
 // ErrGroupNameDup ...
-var ErrGroupNameDup = errors.New("duplicated user group name")
+var ErrGroupNameDup = errors.ConflictError(nil).WithMessage("duplicated user group name")
 
 // Add - Add User Group
 func (d *dao) Add(ctx context.Context, userGroup model.UserGroup) (int, error) {

